src/WebJobs.Extensions.DurableTask/DurableOrchestrationContext.cs

```diff
@@ -438,6 +438,15 @@ public override void ContinueAsNew(object input, bool preserveUnprocessedEvents)
 
                     break;
                 case FunctionType.Orchestrator:
+                    // Instance IDs should not be reused when creating sub-orchestrations. This is a best-effort
+                    // check. We cannot easily check the full hierarchy, so we just look at the current orchestration
+                    // and the immediate parent.
+                    if (string.Equals(instanceId, this.InstanceId, StringComparison.OrdinalIgnoreCase) ||
+                        (this.ParentInstanceId != null && string.Equals(instanceId, this.ParentInstanceId, StringComparison.OrdinalIgnoreCase)))
```

Comment:
Adding a check that the `ParentInstanceId` is not `null` prevents the exception from firing in the scenario where an orchestration without a parent (i.e. a top level orchestration) calls a suborchestration without a specified instance id.